sys/uri_parser/uri_parser.c

```diff
@@ -54,6 +54,11 @@ static char *_consume_scheme(uri_parser_result_t *result, char *uri,
     result->scheme = uri;
     result->scheme_len = p - uri;
 
+    /* check if URI is long enough to contain a '://' */
+    if ((uri_end - p) <= 3) {
```

Comment:
Hmm, could you tie this to the if that follows? It should still be possible to have a URI of the form `urn:` without the `//`. At least, this is how I read https://tools.ietf.org/html/rfc3986#appendix-A

Reply:
Returning immediately with `NULL` would disallow all URIs without `//`?

Could you also add a unit test with your example: `pP://`? (:

Reply:
> Returning immediately with `NULL` would disallow all URIs without `//`?

Good point, I will adjust this by tieing the check to the following if statement. Though I think it would also need an additional check than to ensure that `(uri_end - p) > 1` for `return p + 1`.

> Could you also add a unit test with your example: `pP://`?

Yes, I will :+1:

Reply:
I hope I addressed this :)